rust/gitxetcore/src/merkledb_shard_plumb.rs

```diff
@@ -433,32 +435,43 @@ fn clean_existing_v1_conversions(dir: &Path) -> errors::Result<()> {
     Ok(())
 }
 
-/// Merge Merkledb from notes and upgrade to MDB Shard,
-/// write a conversion meta along with the shard.
-/// Return the meta of the result shard.
-#[allow(dead_code)]
-async fn upgrade_from_v1(
-    config: &XetConfig,
-    output: &Path,
-    notesref: &str,
-) -> errors::Result<MDBShardMeta> {
+/// Merge MerkleMemDB from notes and write its file reconstruction info to a MDBShard.
+/// Upload the converted shard to CAS and register at the shard server.
+/// Write a guard note in V1 refs notes.
+pub async fn upgrade_from_v1_to_v2(config: &XetConfig) -> errors::Result<()> {
+    let repo = GitRepo::open(config.clone())?;
+    let remotes = GitRepo::list_remote_names(repo.repo.clone())?;
+    for r in &remotes {
+        repo.sync_remote_to_notes(r)?;
+    }
+
```

Comment:
We also need to call sync_notes_to_local after this operation, as some of the refs are downloaded to xet_alt.

Reply:
This function actually calls `sync_note_refs_to_local` internally on `merkledb`, `merkledbv2` and `summaries`. This reminds me that [at this place](https://github.com/xetdata/xet-core/blob/7219e58fa6847f9ece4f35e731566501fe176f60/rust/gitxetcore/src/git_integration/git_repo.rs#L315C28-L315C28)  instead of calling `sync_notes_to_dbs_for_implicit_init` we can probably add `sync_note_refs_to_local("reposalt")` to `sync_remote_to_notes`.